op-e2e/actions/l2_proposer.go

```diff
@@ -0,0 +1,78 @@
+package actions
+
+import (
+	"crypto/ecdsa"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum-optimism/optimism/op-node/sources"
+	"github.com/ethereum-optimism/optimism/op-proposer/drivers/l2output"
+)
+
+type ProposerCfg struct {
+	OutputOracleAddr  common.Address
+	ProposerKey       *ecdsa.PrivateKey
+	AllowNonFinalized bool
+}
+
+type L2Proposer struct {
+	log     log.Logger
+	l1      *ethclient.Client
+	driver  *l2output.Driver
+	address common.Address
+	lastTx  common.Hash
+}
+
+func NewL2Proposer(t Testing, log log.Logger, cfg *ProposerCfg, l1 *ethclient.Client, rollupCl *sources.RollupClient) *L2Proposer {
+	chainID, err := l1.ChainID(t.Ctx())
+	require.NoError(t, err)
+	dr, err := l2output.NewDriver(l2output.Config{
+		Log:               log,
+		Name:              "proposer",
+		L1Client:          l1,
+		RollupClient:      rollupCl,
+		AllowNonFinalized: cfg.AllowNonFinalized,
+		L2OOAddr:          cfg.OutputOracleAddr,
+		ChainID:           chainID,
+		PrivKey:           cfg.ProposerKey,
+	})
+	require.NoError(t, err)
+	return &L2Proposer{
+		log:     log,
+		l1:      l1,
+		driver:  dr,
+		address: crypto.PubkeyToAddress(cfg.ProposerKey.PublicKey),
+	}
+}
+
+func (p *L2Proposer) CanPropose(t Testing) bool {
+	start, end, err := p.driver.GetBlockRange(t.Ctx())
+	require.NoError(t, err)
+	return start.Cmp(end) < 0
+}
+
+func (p *L2Proposer) ActMakeProposalTx(t Testing) {
```

Comment:
I don't particularly like how tied the test suite is to the implementation of the proposer. I think it's going to cause more churn in the test framework & it's going to increase the discrepancies of the system we're testing versus the system we want to test.